packages/core/src/routes/interaction/verifications/identifier-payload-verification.ts

```diff
@@ -0,0 +1,107 @@
+import type { Event, SocialConnectorPayload } from '@logto/schemas';
+import type { Provider } from 'oidc-provider';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import { verifyUserPassword } from '#src/lib/user.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import type {
+  PasswordIdentifierPayload,
+  PasscodeIdentifierPayload,
+  InteractionContext,
+  SocialIdentifier,
+  VerifiedEmailIdentifier,
+  VerifiedPhoneIdentifier,
+  AnonymousInteractionResult,
+  PayloadVerifiedInteractionResult,
+  Identifier,
+  AccountIdIdentifier,
+} from '../types/index.js';
+import findUserByIdentifier from '../utils/find-user-by-identifier.js';
+import { isPasscodeIdentifier, isPasswordIdentifier } from '../utils/index.js';
+import { mergeIdentifiers, storeInteractionResult } from '../utils/interaction.js';
+import { verifyIdentifierByPasscode } from '../utils/passcode-validation.js';
+import { verifySocialIdentity } from '../utils/social-verification.js';
+
+const verifyPasswordIdentifier = async (
+  identifier: PasswordIdentifierPayload
+): Promise<AccountIdIdentifier> => {
+  // TODO: Log
+  const { password, ...identity } = identifier;
+  const user = await findUserByIdentifier(identity);
+  const verifiedUser = await verifyUserPassword(user, password);
+
+  const { isSuspended, id } = verifiedUser;
+
+  assertThat(!isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
```

Comment:
Just a reminder: this may also need to be checked for other verification

Reply:
Correct, all the use account existence be verified at the `user-identity-verification` step.  This `payload verification `verifies the payload's valid status only like verified email and phone or social.  

The reason password identifiers check the user's account here is due to its unique form, it is the only way to verify its value is to check the user data. 

The reset guard can be found here. 

<img width="806" alt="image" src="https://user-images.githubusercontent.com/36393111/205852677-d4c8666d-0042-426e-bd0e-79e499ed072a.png">